### PR TITLE
cdemu: update packages

### LINKED ISSUE #342530: Build failure: cdemu
### Steps To Reproduce

1. build cdemu/vhba on latest nixos-unstable

### Build log

```
vhba> /build/vhba-module-20240202/vhba.c:1087:15: error: initialization of 'void (*)(struct platform_device *)' from incompatible pointer type 'int (*)(struct platform_device *)' [-Werror=incompatible-pointer-types]
vhba>  1087 |     .remove = vhba_remove,
vhba>       |               ^~~~~~~~~~~
vhba> /build/vhba-module-20240202/vhba.c:1087:15: note: (near initialization for 'vhba_platform_driver.<anonymous>.remove')
kwin-better-blur> -- Found KF6Config: /nix/store/bkxfqm4zigarz5k4rc58mzmjx964nbbw-kconfig-6.6.0-dev/lib/cmake/KF6Config/KF6ConfigConfig.cmake (found version "6.6.0")
kwin-better-blur> -- Found KF6ConfigWidgets: /nix/store/yin77vn55ngw1dpprm56crzz3mjm6idh-kconfigwidgets-6.6.0-dev/lib/cmake/KF6ConfigWidgets/KF6ConfigWidgetsConfig.cmake (found version "6.6.0")
vhba> cc1: all warnings being treated as errors
kwin-better-blur> -- Found KF6CoreAddons: /nix/store/1kf9xmlm2bgy2jllbms9y3w8vr6b85la-kcoreaddons-6.6.0-dev/lib/cmake/KF6CoreAddons/KF6CoreAddonsConfig.cmake (found version "6.6.0")
vhba> make[3]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/scripts/Makefile.build:244: /build/vhba-module-20240202/vhba.o] Error 1
vhba> make[2]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/Makefile:1926: /build/vhba-module-20240202] Error 2
vhba> make[1]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/Makefile:224: __sub-make] Error 2
vhba> make[1]: Leaving directory '/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build'
vhba> make: *** [Makefile:14: modules] Error 2
error: builder for '/nix/store/9r6hwr3mkcs80hl8gx23d1xpj0y27z42-vhba-20240202.drv' failed with exit code 2;
       last 23 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/xrx8sw2fri58sw11pdqr7k260x77va7d-vhba-module-20240202.tar.xz
       > source root is vhba-module-20240202
       > setting SOURCE_DATE_EPOCH to timestamp 1706870626 of file vhba-module-20240202/vhba.c
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > build flags: SHELL=/nix/store/izpf49b74i15pcr9708s3xdwyqs4jxwl-bash-5.2p32/bin/bash O=\$\(buildRoot\) CC=/nix/store/zznja5f8v3jafffyah1rk46vpfcn38dv-gcc-wrapper-13.3.0/bin/cc HOSTCC=/nix/store/zznja5f8v3jafffyah1rk46vpfcn38dv-gcc-wrapper-13.3.0/bin/cc HOSTLD=/nix/store/b74nxf0yn2dzha02mgdxyklaqjaijzqp-binutils-wrapper-2.42/bin/ld ARCH=x86_64 KDIR=/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build INSTALL_MOD_PATH=\$\(out\)
       > make -C /nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build M=/build/vhba-module-20240202 modules
       > make[1]: Entering directory '/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build'
       >   CC [M]  /build/vhba-module-20240202/vhba.o
       > /build/vhba-module-20240202/vhba.c:1087:15: error: initialization of 'void (*)(struct platform_device *)' from incompatible pointer type 'int (*)(struct platform_device *)' [-Werror=incompatible-pointer-types]
       >  1087 |     .remove = vhba_remove,
       >       |               ^~~~~~~~~~~
       > /build/vhba-module-20240202/vhba.c:1087:15: note: (near initialization for 'vhba_platform_driver.<anonymous>.remove')
       > cc1: all warnings being treated as errors
       > make[3]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/scripts/Makefile.build:244: /build/vhba-module-20240202/vhba.o] Error 1
       > make[2]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/Makefile:1926: /build/vhba-module-20240202] Error 2
       > make[1]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/Makefile:224: __sub-make] Error 2
       > make[1]: Leaving directory '/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build'
       > make: *** [Makefile:14: modules] Error 2
       For full logs, run 'nix log /nix/store/9r6hwr3mkcs80hl8gx23d1xpj0y27z42-vhba-20240202.drv'.
error: 1 dependencies of derivation '/nix/store/5d881pnr6vrz5p14p2dhi7c1vjacyfi9-linux-6.11-modules.drv' failed to build
error: 1 dependencies of derivation '/nix/store/d4w29j0c23kbpkcl4z2ki0grq2idw6cl-nixos-system-nixos-desktop-24.11.20240916.99dc878.drv' failed to build
┏━ 1 Errors:
 ⋮
┃        > unpacking source archive /nix/store/xrx8sw2fri58sw11pdqr7k260x77va7d-vhba-module-20240202.tar.xz
┃        > source root is vhba-module-20240202
┃        > setting SOURCE_DATE_EPOCH to timestamp 1706870626 of file vhba-module-20240202/vhba.c
┃        > Running phase: patchPhase
┃        > Running phase: updateAutotoolsGnuConfigScriptsPhase
┃        > Running phase: configurePhase
┃        > no configure script, doing nothing
┃        > Running phase: buildPhase
┃        > build flags: SHELL=/nix/store/izpf49b74i15pcr9708s3xdwyqs4jxwl-bash-5.2p32/bin/bash O=\$\(buildRoot\) CC=/nix/store/zznja5f8v3jafffyah1rk46vpfcn38dv-gcc-wrapper-13.3.0/bin/cc HOSTCC=/nix/store/zznja5f8v3jafffyah1rk46vp…
┃        > make -C /nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build M=/build/vhba-module-20240202 modules
┃        > make[1]: Entering directory '/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build'
┃        >   CC [M]  /build/vhba-module-20240202/vhba.o
┃        > /build/vhba-module-20240202/vhba.c:1087:15: error: initialization of 'void (*)(struct platform_device *)' from incompatible pointer type 'int (*)(struct platform_device *)' [-Werror=incompatible-pointer-types]
┃        >  1087 |     .remove = vhba_remove,
┃        >       |               ^~~~~~~~~~~
┃        > /build/vhba-module-20240202/vhba.c:1087:15: note: (near initialization for 'vhba_platform_driver.<anonymous>.remove')
┃        > cc1: all warnings being treated as errors
┃        > make[3]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/scripts/Makefile.build:244: /build/vhba-module-20240202/vhba.o] Error 1
┃        > make[2]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/Makefile:1926: /build/vhba-module-20240202] Error 2
┃        > make[1]: *** [/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/source/Makefile:224: __sub-make] Error 2
┃        > make[1]: Leaving directory '/nix/store/d4hzyrkkp1lqy8z30k10q6za7y4gxy5i-linux-6.11-dev/lib/modules/6.11.0/build'
┃        > make: *** [Makefile:14: modules] Error 2
┃        For full logs, run 'nix log /nix/store/9r6hwr3mkcs80hl8gx23d1xpj0y27z42-vhba-20240202.drv'.
```

### Additional context

Looks like there's a new vhba module version: https://sourceforge.net/projects/cdemu/files/vhba-module/

### Notify maintainers
@bendlas 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.10, NixOS, 24.11 (Vicuna), 24.11.20240913.345c263`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/l3amk5lsakpc93him5kry24kax23sn4h-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by bendlas**
## Description of changes

cdemu-daemon: 3.2.6 -> 3.2.7
libmirage: 3.2.7 -> 3.2.9
linuxPackages.vhba: 20240202 -> 20240917

fix https://github.com/NixOS/nixpkgs/issues/342530

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by dotlambda — DISMISSED** [review]
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions

**Comment by keenanweaver** [comment]
Result of `nixpkgs-review pr 342549` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>23 packages built:</summary>
  <ul>
    <li>cdemu-daemon</li>
    <li>image-analyzer</li>
    <li>libmirage</li>
    <li>linuxKernel.packages.linux_4_19.vhba</li>
    <li>linuxPackages_4_19_hardened.vhba (linuxKernel.packages.linux_4_19_hardened.vhba)</li>
    <li>linuxKernel.packages.linux_5_10.vhba</li>
    <li>linuxPackages_5_10_hardened.vhba (linuxKernel.packages.linux_5_10_hardened.vhba)</li>
    <li>linuxKernel.packages.linux_5_15.vhba</li>
    <li>linuxPackages_5_15_hardened.vhba (linuxKernel.packages.linux_5_15_hardened.vhba)</li>
    <li>linuxKernel.packages.linux_5_4.vhba</li>
    <li>linuxPackages_5_4_hardened.vhba (linuxKernel.packages.linux_5_4_hardened.vhba)</li>
    <li>linuxKernel.packages.linux_6_1.vhba</li>
    <li>linuxKernel.packages.linux_6_10.vhba</li>
    <li>linuxPackages_latest.vhba (linuxKernel.packages.linux_6_11.vhba)</li>
    <li>linuxPackages_6_1_hardened.vhba (linuxKernel.packages.linux_6_1_hardened.vhba)</li>
    <li>linuxPackages.vhba (linuxKernel.packages.linux_6_6.vhba)</li>
    <li>linuxPackages_hardened.vhba (linuxPackages_6_6_hardened.vhba)</li>
    <li>linuxPackages_latest-libre.vhba (linuxKernel.packages.linux_latest_libre.vhba)</li>
    <li>linuxPackages-libre.vhba (linuxKernel.packages.linux_libre.vhba)</li>
    <li>linuxPackages_lqx.vhba (linuxKernel.packages.linux_lqx.vhba)</li>
    <li>linuxPackages_xanmod.vhba (linuxKernel.packages.linux_xanmod.vhba)</li>
    <li>linuxPackages_xanmod_latest.vhba (linuxKernel.packages.linux_xanmod_latest.vhba ,linuxPackages_xanmod_stable.vhba)</li>
    <li>linuxPackages_zen.vhba (linuxKernel.packages.linux_zen.vhba)</li>
  </ul>
</details>

**Review by keenanweaver — APPROVED** [review]
Thank you for addressing the issue so quickly!

**Comment by bendlas** [comment]
> https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions

Not applicable, as there are multiple packages with multiple versions. Version numbers are in commit description.

**Comment by dotlambda** [comment]
Excuse me??
Make it one commit per package.

**Comment by SuperSandro2000** [comment]
> Not applicable, as there are multiple packages with multiple versions. Version numbers are in commit description.

For multiple packages there should be multiple commits. That's how we are doing it for years.

**Comment by bendlas** [comment]
> Excuse me?? Make it one commit per package.

Are you serious? Reverting a working commit that fixes a bug out of spite?!

cc @NixOS/moderation please tell em, before I get spicy again

**Comment by dotlambda** [comment]
> I would have squashed the PR anyway, so what's the difference?

Please don't do that. It would render the commit guidelines pointless.

**Comment by bendlas** [comment]
I know that @NixOS/moderation is currently catching up on everything, and I promised to wait for them until I respond further.

But since this is now crossing over into bullying / harrassment / dogpiling again (and I do in fact have emotions as well), I will now lock this thread. Even if you have the ability to circumvent the lock, please take a moment to consider your facts and your goals, before doing so.

**Comment by emilazy** [comment]
While the commit message thing here is a pretty mild violation of our contribution guidelines, and I think @dotlambda’s reaction was excessive, @bendlas’s self‐merge after disregarding the review comments and reaction to pushback, combined with his other recent behaviour, does force me to the conclusion that he is either incapable or unwilling to productively participate collaboratively in our shared project, and that his attitude towards others falls short of what we ought to expect for a committer.

For the benefit of the moderation team, since they have been pinged anyway, I will link recent incidents where I found his behaviour outside of the bounds of acceptable conduct for Nixpkgs (even though I would say the same of some of his interlocutors in them too), though moderators already posted in or were pinged with regards to several of these already:

1. https://discourse.nixos.org/t/can-we-please-stop-breaking-stuff-willy-nilly/48496, https://github.com/NixOS/nixpkgs/pull/318599, https://github.com/NixOS/nixpkgs/pull/331465

1. https://github.com/NixOS/nixpkgs/pull/337478

1. https://github.com/NixOS/nixpkgs/pull/340663#discussion_r1759722787

I am not sure if the jurisdiction here falls properly under the moderation team or the Nixpkgs committer delegation team, but sadly I have found @bendlas to regularly be the centre of deeply unpleasant threads that waste a lot of contributor time and frequently involve him directing flippant snark at others after they escalate to the point of being referred to the moderation team.

I won’t post further in this thread before the moderation team gets to it, as I have no desire to encourage any pile‐ons or get involved in one of these situations again, but given that this is the fourth incident in less than three months I felt it important to point to some of the context here.

**Comment by bendlas** [comment]
Results are in and I have been **formally warned** by the mod team, as a first step towards getting banned.

---

@emilazy I am deeply, **deeply** disappointed that you decided to circumvent the lock, especially since you seem to be looked at as an [arbiter of justice by some](https://github.com/NixOS/nixpkgs/pull/340663#discussion_r1760391565).

I also think that in your list, you may have forgot the one instance, where I complained about a 2-weeks ultimatum being given to maintainers

4. https://github.com/NixOS/nixpkgs/issues/332957#issuecomment-2274824965

I do think I have more than adequate justifications for each of the instances listed, but as I've predicted, [nobody is going to try to hear it](https://github.com/NixOS/nixpkgs/pull/340663#discussion_r1760460578), we're all doing this for fun and in our free time, aren't we?

> @bendlas’s self‐merge

Just out of interest: Is anybody, who's been using this argument aware, that I've not only been the sole maintainer for the cdemu packages, but also its original creator?

In fact, cdemu was the very first thing, that I packaged for NixOS, back when we were still desperate for help. The experience is what gave me confidence enough to go all-in on NixOS, because even as a novice I could create a package with a kernel module, and it's still holding up today!

Since that package is actually the reason I got my maintainer status, I think it will be more than a little poetic if it's also the reason I'm leaving. **That's** the kind of drama, I'm **actually** here for :heart_eyes:

> deeply unpleasant threads that waste a lot of contributor time and frequently involve him directing flippant snark

So many other things to respond to, so little time. I'm picking this, because it perfectly embodies the "whack-a-mole" mentality, that seems to be taking hold in the community: Me being the mole, that dared raise its head and speak.

- Real change is never pleasant. You rustaceans should know by now, or how's that introduction to linux going for you?
- It's hard to stay level-headed, when you're straw-manned to death by [flying monkeys](https://www.verywellmind.com/narcissists-and-flying-monkeys-7552473). I personally think, I've done an above-average job on this, never resorting to personal insults in exchange. may the fates treat you more favorably, when it's your turn in the outgroup.
- I think wasted time can actually be a good thing, if the direction we're going is wrong
  - especially if it's not "wasted" but actually makes people think about what we're doing. I may have failed on that one
- I think flippant snark is a very adequate reaction, when faced with the insinuation that you shouldn't have a voice in the conversation

---

> Make it one commit per package.

@dotlambda you broke it, you fix it. I've re-opened the original issue and put you as owner. I think this is fair, because even my harshest critics seem to agree that using the commit log to carry out a grudge is **way** over the line.

---

Where does this leave us?

I'm not gone yet, but I am also deeply disappointed in the mod team's idea of mediation - just a moderator interrogating me. There was no negotiation with counterparties, no weighing of facts, and the reason I'm the **only** one being formally warned is because apparently "everybody" (?) else apologized (in private? to the mod team?). And now the "mediation" is over and everybody has better things to do :unamused: 

I already knew that there was no appeal process, despite the warning being called "formal" (yes I know, nobody wants to hear it!). It still stings when its your turn. Trust me.

To summarize: The overwhelming trust advance that I held for the mod team is squandered. I kind of feel like a fool now for trolling the RFC-175 folks as hard as I did, just a few weeks ago, and partaking in the laughter when people told me "those never contributed anything to nixos".

---

And to everybody who's been ganging up on me, instead of seeking dialogue: Rest assured that I **will** be more careful with my wording from now on



To everybody at all: My DMs remain open

P.S.:

Please somebody more adept in tone policing tell me if this message is within bounds. I honestly have a hard time telling.

**Comment by picnoir** [comment]
Moderation team hat on: 

Let's not escalate this further.

**Comment by bendlas** [comment]
That's fine, just for completeness sake: to be concluded here https://github.com/NixOS/nixpkgs/pull/342883
